core/io/http_session_manager.hxx

```diff
@@ -204,32 +204,42 @@ class http_session_manager
         std::scoped_lock lock(sessions_mutex_);
         idle_sessions_[type].remove_if([](const auto& s) { return !s; });
         busy_sessions_[type].remove_if([](const auto& s) { return !s; });
-        if (idle_sessions_[type].empty()) {
+        std::shared_ptr<http_session> session{};
+        while (!idle_sessions_[type].empty()) {
+            if (preferred_node.empty()) {
+                session = idle_sessions_[type].front();
+                idle_sessions_[type].pop_front();
+                if (session->reset_idle() != 0) {
+                    break;
```

Comment:
I think this should be continue instead so that it goes back and tries again?

Reply:
Hmmm, if `reset_idle()` is returning 0 that would mean `cancel()` was unable to cancel any operations.  That is the "failure" condition in which we want to try and pick up another session, right?

Reply:
I think this resolved now.